Added unit test Earthworks_01

blaz-p opened the pull request:
# IFC4.3 Unit test

## Unit test: Earthworks_01
This unit test contains an embankment of infrastructure facility using the ifcEarthworksFill concept.

## Review requested
- [IfcEarthworksFill](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifcearthworksfill.htm)
- [IfcEarthworksFillTypeEnum](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifcearthworksfilltypeenum.htm)
- [IfcTriangulatedIrregularNetwork](https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/link/ifctriangulatedirregularnetwork.htm)

@pjanck

pjanck commented on IFC 4.3/Earthworks_01/UT_Earthworks_01.ifc:
```diff
@@ -0,0 +1,65 @@
+#35= IFCSHAPEREPRESENTATION(#13,$,$,(#36));
+#36= IFCTRIANGULATEDIRREGULARNETWORK(#37,$,.F.,((2,144,143),(1,117,116),(18,20,19),(563,907,77),(309,306,307),(24,22,21),(445,443,444),(20,18,17),(42,37,36),(2,162,161),(21,25,24),(33,351,350),(2,166,165),(32,36,33),(2,176,175),(632,941,96),(3,182,181),(17,21,20),(198,195,196),(359,357,358),(3,188,187),(28,26,25),(4,194,193),(44,16,15),(5,203,202),(42,36,17),(7,218,217),(32,30,29),(238,11,12),(52 ...
+#37= IFCCARTESIANPOINTLIST3D(((42459.52618563581,36979.96663112619,74.30073188056490),(42485.72921070703,37021.73076411838,73.85994238924147),(42492.37773898067,37032.32763292525,73.72111812255518),(42499.02626725432,37042.92450173212,73.52832952596715),(42503.72185042583,37050.40863608370,73.35969863252649),(42505.37708576743,37053.04442068565,73.29403191592544),(42505.68920962687,37053.54025718 ...
+#38= IFCGEOGRAPHICELEMENT('071iuxsZ189wc6zewS91Ly',#2,$,$,$,#25,#39,$,.TERRAIN.);
```
In IFC4x3_RC2, `IfcGeographicElement` is deprecated (not noted in documentation, but in its TypeEnum). Use `IfcGeoModel` with `IfcSolidStratum`. Representation with TIN can stay.

AlexBrad1ey replied:
CORRECTION @pjanck 
IfcGeoModel is a geotechnical Assembly and is in essence the same as IfcElementAssembly in it is a aggregator, if the terrain is a singular surface model it should be IfcSolidStratum, if there are multiple layers defined then each layer is a IfcXStratum and grouped into the IfcGeoModel.

pjanck replied:
Thank you @AlexBrad1ey for the correction! I agree wholehaeartedly.

blaz-p replied:
I replaced IfcGeographicElement with IfcSolidStratum, but I don't know where/how to use IfcElementAssembly?

pjanck commented on IFC 4.3/Earthworks_01/readme.md:
```diff
@@ -0,0 +1,50 @@
+- IfcTriangulatedIrregularNetwork
+- IfcCartesianPointList3D
+- IfcEarthworksFillTypeEnum
+- IfcSite
+- IfcProject
+- IfcOwnerHistory
+- IfcPersonAndOgranization
+- IfcPerson
+- IfcOrganization
+- IfcApplication
+- IfcGeometricRepresentationContext
+- IfcAxis2Placement3D
+- IfcCartesianPoint
+- IfcDirection
+- IfcUnitAssignment
+- IfcSIUnit
+- IfcLocalPlacement
+- IfcRelAggregates
+- IfcProductDefinitionShape
+- IfcShapeRepresentation
+- IfcRelContainedInSpatialStructure
```
No need to state these, as they are covered by ProjectSetup_1 and Tin_1.

blaz-p replied:
Deleted in a new commit.

pjanck commented on IFC 4.3/Earthworks_01/readme.md:
```diff
@@ -0,0 +1,50 @@
+- IfcGeographicElement
+- IfcTriangulatedIrregularNetwork
+- IfcCartesianPointList3D
+- IfcEarthworksFillTypeEnum
```
Is deprecated.

blaz-p replied:
IfcGeographicElement was replaced with IfcSolidStratum.

pjanck reviewed (changes requested):
I propose to cojoin all four EarthworksFill's products and geometries into one. TIN as geometry can stay.

AlexBrad1ey commented:
### Advisory

- there are **2 `IfcSites`** is this intensional, IMO multiple sites should only be used for specific use cases or a project working over multiple disjointed areas.

- Your top level facility is an `IfcBuilding` buy the look of the Model this is not really appropriate I would suggest Road or Rail facility or just a generic IfcFacility (dont forget no PredefinedType requires ObjectType to be set)

Other than that this looks fine to me.

blaz-p commented:
Thank you @AlexBrad1ey for your feedback.
Two **IfcSite** were there by mistake. I removed one and replaced **IfcBuilding** with **IfcFacility**.

blaz-p commented:
> ### Advisory
> * there are **2 `IfcSites`** is this intensional, IMO multiple sites should only be used for specific use cases or a project working over multiple disjointed areas.
> * Your top level facility is an `IfcBuilding` buy the look of the Model this is not really appropriate I would suggest Road or Rail facility or just a generic IfcFacility (dont forget no PredefinedType requires ObjectType to be set)
> 
> Other than that this looks fine to me.

pjanck commented:
We're moving to the new repository: https://github.com/bSI-InfraRoom/IFC-infra-unit-test

Please open a new pull request there and close this one.

blaz-p commented:
New pull request was opened in the new repository: https://github.com/bSI-InfraRoom/IFC-infra-unit-test/pull/37.
